### PR TITLE
fix: auto create table

### LINKED ISSUE #846: Automatic table creation bug
### Describe this problem

Automatic table creation has a bug that may cause the first attempt to write data to fail.

### Server version

In PR #844.

### Steps to reproduce

In `integration-test`, `make run-rust`.

### Expected behavior

_No response_

### Additional Information

_No response_

### PR DESCRIPTION
**Pull request by chunshao90**
## Which issue does this PR close?

Closes #846 

## Rationale for this change
Fix bug in #846.

## What changes are included in this PR?
Support automatic table creation in proxy module.

## Are there any user-facing changes?
No.

## How does this change test
Manual test and CI.

### CODE REVIEW

File: integration_tests/cases/env/cluster/ddl/partition_table.sql [hunk 1]
```diff
@@ -42,4 +42,4 @@ SELECT * from partition_table_t where name in ("ceresdb5", "ceresdb6", "ceresdb7
 
 DROP TABLE IF EXISTS `partition_table_t`;
 
-SHOW CREATE TABLE partition_table_t;
+-- SHOW CREATE TABLE partition_table_t;
```
Comment (jiacai2050) [inline integration_tests/cases/env/cluster/ddl/partition_table.sql:45]: Add comments why this is commented out.

File: proxy/src/write.rs [hunk 2]
```diff
@@ -425,6 +416,47 @@ impl<Q: QueryExecutor + 'static> Proxy<Q> {
             })
     }
 
+    async fn handle_auto_create_table(
+        &self,
+        request_id: RequestId,
+        database: &str,
```
Comment (jiacai2050) [inline proxy/src/write.rs:422]: schema

File: proxy/src/write.rs [hunk 3]
```diff
@@ -425,6 +416,47 @@ impl<Q: QueryExecutor + 'static> Proxy<Q> {
...
+                msg: format!("Fail to fetch schema config, schema_name:{database}"),
+            })?
+            .cloned()
+            .unwrap_or_default();
+        for write_table_req in &req.table_requests {
+            let table_info = self
+                .router
+                .fetch_table_info(database, &write_table_req.table)
+                .await?;
+            if table_info.is_some() {
```
Comment (jiacai2050) [inline proxy/src/write.rs:444]: table_info.is_partition_table()

File: proxy/src/write.rs [hunk 4]
```diff
@@ -425,6 +416,47 @@ impl<Q: QueryExecutor + 'static> Proxy<Q> {
...
+                continue;
+            }
+            self.create_table(
```
Comment (jiacai2050) [inline proxy/src/write.rs:447]: This method will always get executed? Even for already existing tables?

Comment (chunshao90) [reply proxy/src/write.rs:447]:
```
if table_info.is_some() {
                continue;
            }
```
Existing tables will skip creation.

**Review by jiacai2050 — APPROVED** [review]
LGTM

**Pull request merged**